### PR TITLE
debug: make toolbar retain x-percent on resize, avoid window control underflow

### LINKED ISSUE #225551: Make the debug-bar right-aligned if its x>50%
Lately I found myself opening two projects and working on them side by side a few times:

image

But when I switch back to one project, the bar is really far in the middle:

image

I can move it back to the far-right:

image

But when I go back to two-project work mode, the bar is hidden:

image

I see three solutions:

* I just deal with it. Moving it back and forth is not the end of the world.
* When the window is resized, the bar keep its current x-percent logic, but stop to the left of the window controls.
* Stop using x-percent and use absolute-left or absolute-right logic instead.

I'm leaning towards the first, but I don't think the x-percent logic works as well now that it can be *literally over the menu bar*. 

image

Food for thought.

### PR DESCRIPTION
**Pull request by connor4312**
Fixes #225551

### CODE REVIEW

File: src/vs/workbench/contrib/debug/browser/debugToolBar.ts [hunk 1]
```diff
@@ -62,6 +64,9 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
 
 	private readonly trackPixelRatioListener = this._register(new MutableDisposable());
 
+	/** Note: changes to this setting require a restarts, so no need to listen to it. */
```
Comment (Tyriar) [inline src/vs/workbench/contrib/debug/browser/debugToolBar.ts:67]:
```suggestion
	/** Note: changes to this setting require a restart, so no need to listen to it. */
```

File: src/vs/workbench/contrib/debug/browser/debugToolBar.ts [hunk 2]
```diff
@@ -188,10 +193,18 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
 		this._register(this.layoutService.onDidChangePartVisibility(() => this.setCoordinates()));
 
 		const resizeListener = this._register(new MutableDisposable());
+		const nextFrameScheduler = this._register(new RunOnceScheduler(() => {
+			const thisWindow = dom.getWindow(this.layoutService.activeContainer);
+			this.setCoordinates(this.getCurrentXPercent() * thisWindow.innerWidth);
+		}, 30));
```
Comment (Tyriar) [inline src/vs/workbench/contrib/debug/browser/debugToolBar.ts:199]: Next frame would be 0, why 30?

Comment (connor4312) [reply src/vs/workbench/contrib/debug/browser/debugToolBar.ts:199]: I actually just wanna make some performance tweaks so this isn't necessary.

**Comment by connor4312** [comment]
actually, with css calculations I can avoid needing to listen to resize at all!

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**